nodeup/pkg/model/secrets.go

```diff
@@ -116,6 +118,44 @@ func (b *SecretBuilder) Build(c *fi.ModelBuilderContext) error {
 		c.AddTask(t)
 	}
 
+	if b.Cluster.Spec.KubeAPIServer.ProxyClientCertFile != nil {
+		cert, err := b.KeyStore.Cert("kube-proxy-client")
```

Comment:
I think we want to write the cert/key always if the `kube-proxy-client` cert/key exists.  I think we should always create the cert/key if the k8s version >= 1.8 (or maybe >= 1.7).  I don't think we should require the ProxyClientCertFile field to be set, in that we can also set it in `nodeup/pkg/model/kubeapiserver.go` (we set e.g. `TLSCertFile` there)